### PR TITLE
Document defaults for stdin, stdout, and stderr methods of Command

### PR DESCRIPTION
**Pull request by mistodon**
For #29370

### CODE REVIEW

File: src/libstd/process.rs [hunk 1]
```diff
@@ -552,6 +552,15 @@ impl Command {
 
     /// Configuration for the child process's standard input (stdin) handle.
     ///
+    /// Defaults to [`inherit`] when used with `spawn` or `status`, and
+    /// defaults to [`piped`] when used with `output`.
+    ///
+    /// On Windows, if the `#![windows_subsystem = "windows"]` attribute is
```
Comment (frewsxcv) [inline src/libstd/process.rs]: can you put the windows-specific lines below a 'Platform-specific behavior' header like we do [here](https://github.com/rust-lang/rust/blob/5f578dfad0dd5d43b28eff71a7e857d10c3f55fe/src/libstd/fs.rs#L1223-L1227)?

File: src/libstd/process.rs [hunk 2]
```diff
@@ -552,6 +552,15 @@ impl Command {
...
+    /// set, no stdin is connected unless explicitly assigned.
```
Comment (frewsxcv) [inline src/libstd/process.rs]: out of curiosity, what do you mean by 'assigned' here? i'm guessing that's some windows-specific concept i'm not familiar with

**Review by frewsxcv — COMMENTED** [review]
looks good! just a couple small comments

**Comment by Pirh** [comment]
It seems I was misunderstanding the Windows-specific behaviour there - it's only relevant when the child process uses the Windows subsystem, not the parent process. (This comes from the comment on #29370 by @retep998 who can maybe give more insight?) I've removed the comments about it in the above commit.

I did some testing and it doesn't look like the child's subsystem affects the defaults of the `stdin`, `stdout`, and `stderr` methods. It seems that, when a child process uses the Windows subsystem, `Stdio::inherit` works differently though. (It will not output to the console, for example, but you can still pipe into and out of it). `Stdio::piped` still works as expected (and so the `output` method behaves identically regardless of the Windows subsystem).

I still don't fully understand this, but it seems like it should be documented on `Stdio::inherit` (and possibly `spawn` and `status`) rather than on `stdin`, `stdout`, `stderr`. I wouldn't know what to write though, so I'll leave that for another PR.

**Comment by carols10cents** [comment]
r? @frewsxcv 

How's this looking, Corey? It looks like your concerns were addressed?

**Comment by frewsxcv** [comment]
i'm not familiar at all with the windows subsystem details regarding this, so i can't be much help there. but this pr looks great, thanks for your contribution!

@bors r+ rollup

**Pull request merged**